internal/monitors/conviva/metricconfig.go

```diff
@@ -0,0 +1,231 @@
+	Filters []string `yaml:"filters"`
+	// MetricLens dimension names. The default is names of all MetricLens dimensions of the account
+	MetricLensDimensions []string `yaml:"metricLensDimensions"`
+	// MetricLens dimension names to exclude.
+	ExcludeMetricLensDimensions []string `yaml:"excludeMetricLensDimensions"`
+	accountID                   string
+	// id:name map of filters derived from the configured Filters
+	filters map[string]string
+	// name:id map of MetricLens dimensions derived from configured MetricLensDimensions
+	metricLensDimensionMap map[string]float64
+	isInitialized          bool
+	// id:name map of filters in filters_warmup status on response
+	filtersWarmup map[string]string
+	// id:name map of filters in filters_not_exist status on response
+	filtersNotExist map[string]string
+	// id:name map of filters in filters_incomplete_data status on response
+	filtersIncompleteData map[string]string
+	mutex                 sync.RWMutex
+}
+
+func (mc *metricConfig) init(service accountsService) {
+	mc.mutex.Lock()
+	defer mc.mutex.Unlock()
+	if !mc.isInitialized {
+		if err := mc.setAccount(service); err != nil {
+			logger.Errorf("Metric %s account setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.setFilters(service); err != nil {
+			logger.Errorf("Metric %s filter(s) setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.setMetricLensDimensions(service); err != nil {
+			logger.Errorf("Metric %s MetricLens dimension(s) setting failure. %+v", mc.Metric, err)
+			return
+		}
+		if err := mc.excludeMetricLensDimensions(service); err != nil {
+			logger.Errorf("Metric %s MetricLens dimension(s) exclusion failure. %+v", mc.Metric, err)
+			return
+		}
+		mc.isInitialized = true
+	}
+}
+
+// setting account id and default account if necessary
+func (mc *metricConfig) setAccount(service accountsService) error {
+	mc.Account = strings.TrimSpace(mc.Account)
+	if mc.Account == "" {
+		if defaultAccount, err := service.getDefault(); err == nil {
+			mc.Account = defaultAccount.Name
+		} else {
+			return err
+		}
+	}
+	var err error
+	if mc.accountID, err = service.getID(mc.Account); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (mc *metricConfig) setFilters(service accountsService) error {
+	if len(mc.Filters) == 0 {
+		mc.Filters = []string{"All Traffic"}
+		if id, err := service.getFilterID(mc.Account, "All Traffic"); err == nil {
+			mc.filters = map[string]string{id: "All Traffic"}
+		} else {
+			return err
+		}
+	} else if strings.TrimSpace(mc.Filters[0]) == "_ALL_" {
+		var allFilters map[string]string
+		var err error
+		if strings.Contains(mc.Metric, "metriclens") {
+			if allFilters, err = service.getMetricLensFilters(mc.Account); err != nil {
+				return err
+			}
+		} else {
+			if allFilters, err = service.getFilters(mc.Account); err != nil {
+				return err
+			}
+		}
+		mc.Filters = make([]string, 0, len(allFilters))
+		mc.filters = make(map[string]string, len(allFilters))
```

Comment:
This is incredibly confusing have two fields that only differ by the initial letter casing.  Can you make the unexported one have a more descriptive name?

Reply:
Say, `filtersByID` or something